lib/chef/knife/s3_source.rb

```diff
@@ -20,11 +20,11 @@ def bucket_obj
       end
 
       def bucket
-        URI(@url).host
+        URI(@url).path.split("/")[1]
```

Comment:
This modification was designed to accept `s3://*`-style URLs. This modification will break that input style. I suggest that if we want to handle `https://`-style URLs (presumably for the purpose of supporting multiple regions), we do some kind of a prefix check on the URL first before attempting to process it.